modules/ROOT/pages/design-common-errors-10.adoc

```diff
@@ -0,0 +1,325 @@
+
+Suppose that the object was defined with a complex property:
+
+----
+user_name:
+    type: object
+    properties:
+        user:
+            name: string
+            lastName: string
+----
+
+The named example would need to look like this:
+
+----
+#%RAML 1.0 NamedExample
+MyExampleName:
+    user:
+        name: Lionel
+        lastName: Ma
+----
+
+Suppose that instead it looked like this:
+
+----
+#%RAML 1.0 NamedExample
+user:
+    name: Lionel
+    lastName: Ma
+----
+
+The parser would assume that `user` was the name of the example and that the object had two properties, not one complex property.
+
+== Common Error 4
+// APIMF-966
+*_Not using a correct value for the `protocols` node_*
+
+The value of the `protocols` node must be an array. The array must be:
+
+* `[http]` or `[HTTP]`
+* `[https]` or `[HTTPS]`
+* `[http, https]` or `[HTTP, HTTPS]`
+
+== Common Error 5
+// APIMF-967
+*_Including an example response that contains invalid JSON_*
+
+When a JSON file is included as the example of a response message, the JSON in the file must be valid. In this example of the error, the example of the response for the 200 response code contains an `!include` statement. The JSON in the included file incorrectly contains a comma after the last key/value pair.
+
+----
+#%RAML 1.0
+title: ExampleRAML
+...
+/resume:
+  description: "Gets candidate's resume."
+  get:
+    queryParameters:
+       ...
+    headers:
+      ...
+    responses:
+      200:
+        body:
+          application/json:
+            example: !include exampleResumeData-200.json
+      500:
+        ...
+----
+
+
+----
+{
+...
+"assesments.characteristic.focusofattention.data"= "",
+}
+
+
+----
+
+
+
+== Common Error 6
+// APIMF-1030
+*_Referencing libraries by using the `type` key_*
+
+As explained in the RAML 1.0 specification, you must apply libraries with the `uses` node:
+
+____
+Any number of libraries can be applied by using the OPTIONAL uses node ONLY at the root of a ["master"] RAML or RAML fragment file. The value of the uses node is a map of key-value pairs. The keys are treated as library names, or namespaces, and the value MUST be the location of a RAML library file, usually an external RAML library fragment document.
+____
+
+Therefore, the following snippet is incorrect, given that the file `financeDetail.raml` is a library.
+
+----
+#%RAML 1.0
+title: ExampleRAML
+...
+/claims:
+  /{claim-id}:
+    patch:
+      body:
+        application/json:
+          type: !include financeDetail.raml
+----
+
+This next snippet is correct.
+
+----
+#%RAML 1.0
+title: ExampleRAML
+...
+/claims:
+  /{claim-id}:
+    patch:
+      body:
+        application/json:
+          uses:
+            myLibrary: financeDetail.raml
+----
```

Comment:
The uses entry needs to be defined at the api root level. 
The library can declares more than one element (like types, traits, etc). Each one of those elements MUST be reference under the library alias from the api.

#%RAML 1.0
title: ExampleRAML
uses:
    lib: library.raml
/claims:
  /{claim-id}:
    patch:
      body:
        application/json:
          type: lib.myType

Reply:
So, if the library is the file `financeDetail.raml', would this be the correct example?
```
#%RAML 1.0
title: ExampleRAML
uses:
  lib: financeDetail.raml
/claims:
  /{claim-id}:
    patch:
      body:
        application/json:
          type: lib.myType
```

Reply:
Fixed.